code/controlconfig/controlsconfig.h

```diff
@@ -561,7 +561,7 @@ class CC_preset {
 public:
 	SCP_vector<CCB> bindings;
 	SCP_string name;
-	Preset_t type;
+	Preset_t type = Preset_t::hardcode;
```

Comment:
Please omit this change and leave it uninitialized, so that Coverity, compiler, or another tool may be able to find mistakes where the type wasn't set at creation.

As uh, I hate to admit it, but `CC_preset` wasn't fully OOP compliant in its implementation.  `CC_preset::type` should be a constant, since it can not change between `hardcode`, `tbl` (.tbl entry), or `pst` (preset file).  `hardcode` and `tbl` presets obviously can not be modified or created during game execution.  Preset files can be created, edited, and potentially deleted in-game, so having `CC_preset::type` as a variable made copying from the other two types convenient, but nevertheless leaves behind a code stink.

So uhm, yeah, a bit of a refactor is needed at some point, but I believe its low-priority.

Reply:
oh its already merged. wellok.   It'll be fine. 🤞

Reply:
Coverity is only smart enough to detect values that are uninitialized at creation.  It can't detect errors caused by uninitialized values somewhere down the pipeline.  That would require a memory analysis debugger, which is both trickier to use and would require a dedicated search effort.

Putting a refactor on the to-to list is a better solution, and a good idea.